### PR TITLE
Use TabNet for tabular classification

### PR DESCRIPTION
**Pull request by teddykoker**
## What does this PR do?

Uses tabnet for tabular classification

### CODE REVIEW

**Comment by teddykoker** [comment]
The only failing tests are the model download (since the tabular model has changed) I will update the weights in S3 when the PR is accepted

**Review by tchaton — APPROVED** [review]
Excellent !

File: flash/tabular/classification/model.py [hunk 1]
```diff
@@ -93,16 +84,9 @@ def predict(
...
-            # concatenate embeddings for each categorical variable
-            x = [e(x_cat[:, i]) for i, e in enumerate(self.embs)]
-            x = torch.cat(x, dim=1)
-        if self.bn_num is not None:
-            x_num = self.bn_num(x_num)
-            x = torch.cat([x_num, x], dim=1) if len(self.embs) else x_num
-        x = self.model(x)
-        return x
+        x = torch.cat([x for x in x_in if x.numel()],
+                      dim=1)  # TabNet takes single input, x_in is composed of (categorical, numerical)
```
Comment (Borda) [inline flash/tabular/classification/model.py:88]:
```suggestion
        # TabNet takes single input, x_in is composed of (categorical, numerical)
        x = torch.cat([x for x in x_in if x.numel()], dim=1)
```

File: flash/vision/backbones.py [hunk 2]
```diff
@@ -47,7 +47,7 @@ def bolts_backbone_and_num_features(model_name: str) -> Tuple[nn.Module, int]:
     >>> bolts_backbone_and_num_features('simclr-imagenet')  # doctest: +ELLIPSIS
     (Sequential(...), 2048)
     >>> bolts_backbone_and_num_features('swav-imagenet')  # doctest: +ELLIPSIS
-    (Sequential(...), 3000)
+    (Sequential(...), 2048)
```
Comment (Borda) [inline flash/vision/backbones.py:50]: this has been fixed already, right?

**Review by Borda — APPROVED** [review]

File: flash/vision/backbones.py (hunk 2 above)
Comment (teddykoker) [reply flash/vision/backbones.py:50]: Forgot to fix the doctest, in the last PR, for some reason tests still passed then but not now

**Review by edenlightning — APPROVED** [review]

**Pull request merged**